store/details_v2.go

```diff
@@ -174,6 +174,9 @@ func copyNonZeroFrom(src, dst *storeSnap) {
 	}
 	if src.Download.URL != "" {
 		dst.Download = src.Download
+	} else if src.Download.Size != 0 {
```

Comment:
(nitpick) do we really need the if/else here? Could we simply copy the size if there is a size?